src/test/java/com/exadel/demo/service/UserServiceTest.java

```diff
@@ -0,0 +1,86 @@
+package com.exadel.demo.service;
+
+import com.exadel.demo.dto.RoleDto;
+import com.exadel.demo.dto.UserDto;
+import com.exadel.demo.entity.Role;
+import com.exadel.demo.entity.User;
+import com.exadel.demo.repository.UserRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.modelmapper.ModelMapper;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class UserServiceTest {
+    @Mock
+    private UserRepository userRepository;
+    @Mock
+    private ModelMapper modelMapper;
+
+    private User user;
+
+    private UserDto userDto;
+
+    private UserService userService;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+
+        userService = new UserService(userRepository, modelMapper);
+        user = new User("", "", "", new Role());
+        User savedUser = new User("", "", "", new Role());
```

Comment:
You can have this object like class filed and maybe later we can use it in other tests